pkgs/games/fltrator/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchurl, unzip, fltk, which, libjpeg }:
+
+stdenv.mkDerivation rec {
+  name = "fltrator-${version}";
+  version = "2.3";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/fltrator/fltrator-${version}-code.zip";
+    sha256 = "125aqq1sfrm0c9cm6gyylwdmc8xrb0rjf563xvw7q28sdbl6ayp7";
+  };
+
+  buildInputs = [ fltk libjpeg ];
+  nativeBuildInputs = [ unzip which ];
+
+  postPatch = ''
+    substituteInPlace src/fltrator.cxx\
+      --replace 'home += "fltrator/"' "home = \"$out/fltrator/\""
+    substituteInPlace src/fltrator-landscape.cxx\
+      --replace 'home += "fltrator/"' "home = \"$out/fltrator/\""
+    substituteInPlace rsc/fltrator.desktop \
+      --replace 'Exec=fltrator' "Exec=$out/bin/fltrator" \
+      --replace 'Icon=fltrator2' "Icon=$out/fltrator/icon.png"
+  '';
+
+  dontAddPrefix = true;
+
+  makeFlags = [ "HOME=$(out)" "RSC_PATH=$(out)/fltrator"];
+
+  postInstall = ''
+    mkdir -p $out/share/applications
+    cp rsc/fltrator.desktop $out/share/applications
+    cp rsc/fltrator-128.png $out/fltrator/icon.png
```

Comment:
Icons should go to `$out/share/icons/hicolor/128x128/apps` as per [icon-theme-spec](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html).